cishouseholds/derive.py

```diff
@@ -16,6 +16,17 @@
 from cishouseholds.pyspark_utils import get_or_create_spark_session
 
 
+def assign_date_from_filename(df: DataFrame, column_name_to_assign: str, filename_column: str):
+    """
+    Populate a pyspark date column with the date contained in the filename column
+    """
+    df = df.withColumn(
+        column_name_to_assign,
+        F.to_date(F.regexp_extract(F.col(filename_column), r"(\d{8})(_\d{4})?(.csv)", 1), format="yyyyMMdd"),
+    )
```

Comment:
Also, could you update this to catch the whole datetime instead please? We should use `000000` as a default when no time can be extracted from the filename (i.e. pre-digital files)